### PR TITLE
Fix parsing raygui comments and vector2 enum

### PR DESCRIPTION
**Pull request by IrishBruse**
image
image

### CODE REVIEW

File: src/extras/raygui.h [hunk 1]
```diff
@@ -239,7 +239,7 @@
         float y;
     } Vector2;
 
-    // Vector3 type                 // -- ConvertHSVtoRGB(), ConvertRGBtoHSV()
```
Comment (IrishBruse) [inline src/extras/raygui.h:242]: Not sure why this was here in the comment

File: src/extras/raygui.h [hunk 2]
```diff
@@ -593,9 +593,7 @@ RAYGUIAPI bool GuiCheckIconPixel(int iconId, int x, int y);     // Check icon pi
 // Number of elemens depend on RICON_SIZE (by default 16x16 pixels)
 #define RICON_DATA_ELEMENTS   (RICON_SIZE*RICON_SIZE/32)
 
-//----------------------------------------------------------------------------------
 // Icons enumeration
-//----------------------------------------------------------------------------------
 typedef enum {
```
Comment (IrishBruse) [inline src/extras/raygui.h:597]: the //----- confuse the parser when gening a description

File: src/extras/raygui.h [hunk 3]
```diff
@@ -1141,7 +1139,12 @@ static unsigned int guiIcons[RICON_MAX_ICONS*RICON_DATA_ELEMENTS] = {
 // Types and Structures Definition
 //----------------------------------------------------------------------------------
 // Gui control property style color element
-typedef enum { BORDER = 0, BASE, TEXT, OTHER } GuiPropertyElement;
```
Comment (IrishBruse) [inline src/extras/raygui.h:1144]: this need to be multiline for the parser to read it correctly

File: parser/raylib_parser.c [hunk 4]
```diff
@@ -241,7 +241,7 @@ int main(int argc, char* argv[])
     for (int i = 0; i < linesCount; i++)
     {
         // Read enum line
-        if (IsTextEqual(lines[i], "typedef enum {", 14))
+        if (IsTextEqual(lines[i], "typedef enum {", 14) && lines[i][TextLength(lines[i])-1] != ';') // ignore inline enums
```
Comment (IrishBruse) [inline parser/raylib_parser.c:244]: this will ignore the c++ bool enum that is defined

**Comment by raysan5** [comment]
Note that `raygui.h` version in raylib repo is not the last one, `raygui.h` changes should be done in [`raygui` repo](https://github.com/raysan5/raygui). Also note that raygui changes frequently.

**Comment by IrishBruse** [comment]
Ok moved the fixes to the raygui repo

**Pull request merged**